### PR TITLE
[batch] change select for update to select lock in share mode

### PR DESCRIPTION
**Pull request by jigold**
I used FOR UPDATE when I should have used LOCK IN SHARE MODE.

I added the diff between initial.sql and the new migration at the top. Not sure if this is helpful or what the standard for figuring out how to track changes should be.

### CODE REVIEW

**Comment by jigold** [comment]
I'm not sure the diff is a good idea.

File: batch/sql/add-lock-in-share-mode.sql [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+  SET delta_cores_mcpu = IFNULL(delta_cores_mcpu, 0);
+
+  SET attempt_exists = EXISTS (SELECT * FROM attempts
+                               WHERE batch_id = in_batch_id AND
+                                 job_id = in_job_id AND attempt_id = in_attempt_id
+                               FOR UPDATE);
+
+  IF NOT attempt_exists AND in_attempt_id IS NOT NULL THEN
+    INSERT INTO attempts (batch_id, job_id, attempt_id, instance_name) VALUES (in_batch_id, in_job_id, in_attempt_id, in_instance_name);
+    SELECT state INTO cur_instance_state FROM instances WHERE name = in_instance_name LOCK IN SHARE MODE;
```
Comment (cseed) [inline batch/sql/add-lock-in-share-mode.sql]: This change seems wrong.  You update instances two lines below.  Why should this be in SHARE MODE?

Comment (jigold) [reply batch/sql/add-lock-in-share-mode.sql]: good catch!

File: batch/sql/add-lock-in-share-mode.sql [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+  SELECT end_time INTO cur_end_time FROM attempts
+  WHERE batch_id = in_batch_id AND job_id = in_job_id AND attempt_id = in_attempt_id
+  FOR UPDATE;
+
+  UPDATE attempts
+  SET end_time = new_end_time, reason = new_reason
+  WHERE batch_id = in_batch_id AND job_id = in_job_id AND attempt_id = in_attempt_id;
+
+  SELECT state INTO cur_instance_state FROM instances WHERE name = in_instance_name LOCK IN SHARE MODE;
```
Comment (cseed) [inline batch/sql/add-lock-in-share-mode.sql]: This one also updates just below.

File: batch/sql/add-lock-in-share-mode.sql [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+  SELECT end_time INTO cur_end_time FROM attempts
+  WHERE batch_id = in_batch_id AND job_id = in_job_id AND attempt_id = in_attempt_id
+  FOR UPDATE;
+
+  UPDATE attempts
+  SET start_time = new_start_time, end_time = new_end_time, reason = new_reason
+  WHERE batch_id = in_batch_id AND job_id = in_job_id AND attempt_id = in_attempt_id;
+
+  SELECT state INTO cur_instance_state FROM instances WHERE name = in_instance_name LOCK IN SHARE MODE;
```
Comment (cseed) [inline batch/sql/add-lock-in-share-mode.sql]: And this one.

**Pull request merged**